src/modules/esl-utils/dom/scroll/utils.ts

```diff
@@ -51,4 +52,44 @@ export abstract class ScrollUtils {
     initiator && initiatorSet.delete(initiator);
     (initiatorSet.size === 0) && ScrollUtils.unlock();
   }
+
+  private static _onScrollElement(target: Element): any {
+    const scrollTop = target.scrollTop;
+    const scrollLeft = target.scrollLeft;
+    return (e: KeyboardEvent) => {
+      if (e.target === target) {
+        target.scrollTop = scrollTop;
+        target.scrollLeft = scrollLeft;
+      }
+    };
+  }
+
+  private static _onScrollWindow(): any {
+    const scrollTop = window.scrollY || window.pageYOffset;
+    const scrollLeft = window.scrollX || window.pageXOffset;
+    return () => window.scrollTo(scrollLeft, scrollTop);
+  }
```

Comment:
That's fine and it almost works (if we do not include artifacts that come from the scroll passive event nature in the modern browsers)
But there is a problem with the main goal of this issue:
The business's purpose was to create a mechanism that allows us to lock user manipulations but let us scroll the container via JS (the ScrollTo enhancement story...).